### PR TITLE
Check frame conditions before __CPROVER_havoc_object

### PR DESCRIPTION
**Pull request by feliperodri**
Signed-off-by: Felipe R. Monteiro <felisous@amazon.com>

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-instrument/contracts/contracts.cpp [hunk 1]
```diff
@@ -748,7 +741,8 @@ void code_contractst::instrument_call_statement(
     called_name =
       to_symbol_expr(instruction_iterator->call_function()).get_identifier();
   }
-
+  log.warning() << "called function: " << id2string(called_name)
+                << messaget::eom;
```
Comment (SaswatPadhi) [inline src/goto-instrument/contracts/contracts.cpp]: Is this some leftover debugging code?

File: src/goto-instrument/contracts/contracts.cpp [hunk 2]
```diff
@@ -972,9 +953,33 @@ void code_contractst::check_frame_conditions(
...
+      add_containment_check(program, assigns, instruction_it, havoc_argument);
+      assigns.remove_from_local_write_set(havoc_argument);
+      assigns.remove_from_global_write_set(havoc_argument);
+    }
   }
 }
 
+void code_contractst::add_containment_check(
+  goto_programt &program,
+  assigns_clauset &assigns,
```
Comment (SaswatPadhi) [inline src/goto-instrument/contracts/contracts.cpp]:
```suggestion
  const assigns_clauset &assigns,
```

**Review by SaswatPadhi — APPROVED** [review]
LGTM! Only minor comments below.

File: src/goto-instrument/contracts/contracts.cpp (hunk 1 above)
Comment (feliperodri) [reply src/goto-instrument/contracts/contracts.cpp]: Yes, removed it.

File: src/goto-instrument/contracts/contracts.cpp (hunk 2 above)
Comment (feliperodri) [reply src/goto-instrument/contracts/contracts.cpp]: Implemented as suggested.

**Pull request merged**